api/segments/models.py

```diff
@@ -181,9 +183,18 @@ def does_identity_match(
 
         for trait in traits:
             if trait.trait_key == self.property:
-                if trait.value_type == INTEGER:
+                if self.operator == MODULO:
+                    if trait.value_type in [INTEGER, FLOAT]:
+                        return self._check_modulo_operator(
+                            trait.integer_value
```

Comment:
We're kind of rewriting the logic contained in the `get_trait_value()` (or `trait_value` property) here. I think you could simplify this to remove the conditional logic here.